modules/java/native/java_machine.c

```diff
@@ -171,7 +174,7 @@ _jvm_options_split(const gchar *jvm_options_str)
           continue;
         }
 
-      jvm_options_array = _jvm_options_array_append(jvm_options_array, options_str_array[i]);
+      jvm_options_array = _jvm_options_array_append(jvm_options_array, g_strdup(options_str_array[i]));
```

Comment:
I added a refactor patch to the top of the patchset. Please check!

Reply:
We concluded to retain the original version. It looks strange to strup an strdupped string, and free both during deinit. These kind of strangenesses could be resolved by introducing extra logic into the jvm_array architecture, like adding key=value type parameters, single parameters (... any other types?). For now this would be an overkill. I dropped the patch.